src/server/controllers/index.js

```diff
@@ -0,0 +1,84 @@
+const  express = require('express');
+var router = express.Router();
+
+/* Require model */
+var models = require('../models');
+/* Require middlewares functions */
+
+router.get('/', function (req, res) {
+    res.render('index.html', {title: "Interview Coding Challenge"});
+});
+
+router.get('/getAllFavStuffs', function (req, res) {
+    models.favorite_stuff.find({}, function (err, data) {
+        if (err) {
+            res.json({error: true, message: "There isn't information yet.", response: ""});
+        }
+        res.json({error: false, message: "All Favorites Stuff", response: data});
+    });
+});
+
+router.get('/getAllFavStuffsNames', function (req, res) {
+    models.favorite_stuff.find({}, {fav_stuff_name: 1}, function (err, data) {
+        if (err) {
+            res.json({error: true, message: "There isn't information yet.", response: ""});
+        }
+        res.json(data);
+    });
+});
+
+router.post('/addNewFavStuff*', function (req, res) {
+    if ((typeof req.body.fav_stuff_name == "undefined") || (typeof req.body.fav_stuff_list == "undefined")) {
```

Comment:
Good validation. Though, I was able to pass:

![image](https://user-images.githubusercontent.com/601266/27670904-ad69bc5a-5c55-11e7-96d7-09a8f3ee7258.png)

Hence, my list now looks like:

![image](https://user-images.githubusercontent.com/601266/27670914-bdbde46e-5c55-11e7-8c6b-88071af3a0f6.png)